third_party/antlr4/runtime/Cpp/cmake/ExternalAntlr4Cpp.cmake

```diff
@@ -51,7 +51,7 @@ endif()
 if(${CMAKE_GENERATOR} MATCHES ".* Makefiles")
   # This avoids
   # 'warning: jobserver unavailable: using -j1. Add '+' to parent make rule.'
-  set(ANTLR4_BUILD_COMMAND $(MAKE))
+  set(ANTLR4_BUILD_COMMAND $(MAKE) CXX_FLAGS="-fPIC" C_FLAGS="-fPIC")
```

Comment:
Changes in these files feels wrong.  Should these go upstream first?  I feel like something is missing.

Reply:
This is a slightly strange way to build this, because I want to statically link one library to another shared library to produce a single shared library. The right way is to probably modify this file since this is a unique situation.

It would be more common to either package the shared libraries, or statically link them into an executable. So I don't expect this would get merged upstream.

Reply:
I could copy this somewhere outside `third_party/` if that's preferred.

Reply:
I believe if you proposed something akin to the option `ANTLR4_WITH_STATIC_CRT` (see line 74-75) about enabling PIC even when building a static library, it would at a minimum start a valuable conversation with the ANTLR4 team and potentially be upstreamed.  I think that approach will pull in insight from the ANTLR4 folks, and lead to a better solution.

Reply:
This will take time. I've created [an issue](https://github.com/SymbiFlow/fasm/issues/42), but I'd like to merge this for now.

Reply:
I think taking that time now is more valuable than landing this PR immediately.

Reply:
@mithro What are your thoughts on this? I'd like to get this this week.

Reply:
This issue appears to have occurred in at least two other instances with ANTLR4:
 - https://github.com/antlr/antlr4/issues/2688
 - https://github.com/antlr/antlr4/issues/2776

Rather than maintain a local fix, we should get the fix in upstream.

Reply:
I agree, but I don't think it needs block the work we've done on FASM being used today, so I've filed an issue.

Reply:
I found a simple solution that doesn't require changing these files - just set CFLAGS/CXXFLAGS to `-fPIC`.